Lib/http/server.py

```diff
@@ -763,9 +766,30 @@ def send_head(self):
                             f.close()
                             return None
 
-            self.send_response(HTTPStatus.OK)
+            if self.range:
+                start, end = self.range
+                if start is None:
+                    # `end` here means suffix length
+                    start = max(0, fs.st_size - end)
+                    end = fs.st_size - 1
+                if start >= fs.st_size:
+                    # 416 REQUESTED_RANGE_NOT_SATISFIABLE means that none of the range values overlap the extent of the resource
+                    f.close()
+                    self.send_error(HTTPStatus.REQUESTED_RANGE_NOT_SATISFIABLE)
+                    return None
+                if end is None or end >= fs.st_size:
+                    end = fs.st_size - 1
+                self.send_response(HTTPStatus.PARTIAL_CONTENT)
+                self.send_header("Content-Range", "bytes %s-%s/%s" % (start, end, fs.st_size))
+                self.send_header("Content-Length", str(end - start + 1))
+
+                # Update range to be sent to be used later in copyfile
+                self.range = (start, end)
+            else:
+                self.send_response(HTTPStatus.OK)
+                self.send_header("Accept-Ranges", "bytes")
```

Comment:
Is this one needed?

Reply:
I think this is correct. The server need to respond `accept-ranges: bytes` and `content-length: <num>` to tell the client it supports this feature.

Reply:
What I mean is that this wasn't the case until now. So I'm wondering whether this additional line is needed.

Reply:
According to [RFC 7233](https://datatracker.ietf.org/doc/html/rfc7233#section-2.3), http server MAY send a `Accept-Ranges` header. It's not mandatory, but indicates the server support range headers.

Reply:
If this can be used to indicate the range support, then let's keep it. Should it be sent when handling the range as well? (namely for the header to be present whether a range is specified or not)?